### PR TITLE
feat(COM-1348) Replace showClearButton boolean property in favor of clearButtonMode (item-selected, has-value, never, always) to determine the criteria/visibility of the clear button

### PR DESCRIPTION
**Pull request by wescosta**
**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

File: packages/components/src/components/form/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -56,15 +56,25 @@ export const Autocomplete: FC<AutocompleteProps> & AutocompleteStaticMembers = (
   );
 };
 
-const AutocompleteInput: FC<AutocompleteInputProps> = ({ showClearButton, ...rest }) => {
+const AutocompleteInput: FC<AutocompleteInputProps> = ({
+  showClearButton,
+  showClearButtonWhen = 'item-selected',
```
Comment (guilean) [inline packages/components/src/components/form/Autocomplete/Autocomplete.tsx:61]: Would you consider other namings like `clearButtonMode` or `clearButtonCriteria` ?

Comment (guilean) [reply packages/components/src/components/form/Autocomplete/Autocomplete.tsx:61]: Maybe we should consider removing the `showClearButton` and have an enum for the `clearButtonCriteria` or `clearButtonMode` with values:

- `item-selected`: Show the clear button when an item is selected.
- `has-value`: Show the clear button when there is input value.
- `never`: Never show the clear button.

Comment (wescosta) [reply packages/components/src/components/form/Autocomplete/Autocomplete.tsx:61]: Yeah. I've thought about removing it, but removing a prop would be a breaking change.

Should we still do it? WDYT?

Comment (guilean) [reply packages/components/src/components/form/Autocomplete/Autocomplete.tsx:61]: I'd prefer to have just one property to deal with. Let's do a major

**Review by guilean — APPROVED** [review]

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**